library/src/main/java/org/apache/apex/malhar/lib/state/managed/IncrementalCheckpointManager.java

```diff
@@ -109,8 +110,8 @@ public void run()
           transferWindowFiles();
           if (latestExpiredTimeBucket.get() > -1) {
             try {
-              managedStateContext.getBucketsFileSystem().deleteTimeBucketsLessThanEqualTo(
-                  latestExpiredTimeBucket.getAndSet(-1));
+              latestPurgedTimeBucket = latestExpiredTimeBucket.getAndSet(-1);
```

Comment:
Each streaming window records its own lateness horizon. The purge happens only if the time bucket is beyond the lateness horizon at the committed streaming window, so we never need the time bucket that is purged.